src/async-wrap.cc

```diff
@@ -448,6 +452,56 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
   }, static_cast<void*>(env));
 }
 
+static void AsyncTaskScheduled(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString())
+    return env->ThrowTypeError("first argument must be a string");
+  if (!args[1]->IsNumber())
+    return env->ThrowTypeError("second argument must be a number");
+  if (!args[2]->IsBoolean())
+    return env->ThrowTypeError("third argument must be a boolean");
+
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  intptr_t taskId = args[1].As<Integer>()->Value();
+  void* task = reinterpret_cast<void*>(taskId);
+  bool recurring = args[2].As<Boolean>()->Value();
+
+  env->inspector_agent()->AsyncTaskScheduled(taskNameView, task, recurring);
+}
+
+static void AsyncTaskCanceled(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskCanceled(reinterpret_cast<void*>(taskId));
+}
+
+static void AsyncTaskStarted(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskStarted(reinterpret_cast<void*>(taskId));
+}
+
+static void AsyncTaskFinished(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskFinished(reinterpret_cast<void*>(taskId));
+}
```

Comment:
You could DRY these a little by using a template function.  As well, this probably needs to be inside a `#if HAVE_INSPECTOR` guard.  `./configure --without-inspector` should build/work.